### PR TITLE
fix: react-combobox activedescendant calculation updates

### LINKED ISSUE #26316: [Bug]: Combobox - filtering - set focus into list when user deletes wrong letters
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 Intel(R) Xeon(R) E-2176M  CPU @ 2.70GHz
    Memory: 12.51 GB / 31.74 GB
  Browsers:
    Edge: Spartan (44.22621.1105.0), Chromium (108.0.1462.76)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?
yes

### Reproduction

https://codesandbox.io/s/19z9eh?file=/example.tsx

### Bug Description
Issue is reported for filtering scenario.

Steps:
1. Screen reader is on (tested with Chrome/NVDA)
2. Navigate to combobox
3. Type string "hamstee"
4. delete the last letter "e" with backspace 
5. observe behavior 
## Actual Behavior
list is opened, but focus stays in the dropdown input

## Expected Behavior
focus is set into list

Comment: once I delete one more letter "e" then focus was set into list

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### LINKED ISSUE #26359: [Bug]: Combobox - Default - Screen reader doesn't narrate letter in input field combobox input
### Library

React Northstar / v0 (@fluentui/react-northstar)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 Intel(R) Xeon(R) E-2176M  CPU @ 2.70GHz
    Memory: 12.51 GB / 31.74 GB
  Browsers:
    Edge: Spartan (44.22621.1105.0), Chromium (108.0.1462.76)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/s/7mzpld?file=/example.tsx

### Bug Description

**Navigate between letters:**
Steps:
1. Run JAWS or NVDA
2. Navigate to combobox 
3. Write existing option like "Hamster"
4. Navigate in the list pressing up/down arrow keys
5.  press left/right arrow keys to navigate between the letters which user typed in input 
6. observe screen reader behavior

## Actual Behavior
screen reader is not narrating letters 

## Expected Behavior
screen reader narrate letters which user navigate on 

**Delete letters:**
Steps:
1. Run JAWS or NVDA
2. Navigate to combobox 
3. Write existing option like "Hamster"
4. Navigate in the list pressing up/down arrow keys
5.  press backspace to delete last letter
6. press backspace again to delete one more letter
7. observe screen reader behavior

## Actual Behavior
screen reader is not narrating deleted letters

## Expected Behavior
screen reader narrate letters which user deleted

Comment: 
It works with ARIA example:
https://w3c.github.io/aria-practices/examples/combobox/combobox-autocomplete-list.html
Looks like issue is when we use left/right arrow keys we need to get screen reader focus on input. The similar with backspace, in order enable screen reader to narrate deleted letters we need to have screen reader focus on input. 
At least this is what I observe on ARIA example that screen reader focus ring moving to input when user navigate between letters or preses backspace. 

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by smhigley**
There were two behavior issues with the `aria-activedescendant` calculation in combobox:

1. When the author mutated Option children while open, the active option was not recalculated

Now, when the children are changed while open and there is no current active option, combobox resets the active option.
Fixes #26316

2. A screen reader bug prevents the input value from being read while an `aria-activedescendant` association is present for JAWS & NVDA

While this should work out of the box, there's also a relatively easy fix we can do -- `aria-activedescendant` attribute (but not internal state) is cleared with left/right arrows, and it is restored with any other keyboard interaction. Home/end do not apply here, since they do listbox interactions. I left a comment by the relevant code, so if the screen reader bug is fixed, we can remove it.

Fixes #26359

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                             |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                  Change |
| :---------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `80.888 kB`<br />`25.842 kB` | `81.019 kB`<br />`25.913 kB` | `131 B` <br />`71 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/ComboboxField.fixture.js'>ComboboxField</abbr>                    | `78.774 kB`<br />`25.985 kB` |  `78.901 kB`<br />`26.05 kB` | `127 B` <br />`65 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `79.674 kB`<br />`25.637 kB` | `79.687 kB`<br />`25.649 kB` |  `13 B` <br />`12 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `59.888 kB`<br />`16.652 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `187.539 kB`<br />`52.638 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.379 kB`<br />`11.322 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.069 kB`<br />`2.053 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/294d8a773e2d154f04addf24ff8600a7bb408a31'>294d8a773e2d154f04addf24ff8600a7bb408a31</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/Avatar-mount_base.html">1150</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/Avatar-mount.html">1219</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/Button-mount_base.html">864</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/Button-mount.html">860</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/FluentProvider-mount_base.html">1570</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/FluentProvider-mount.html">1549</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/FluentProviderWithTheme-mount_base.html">627</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/FluentProviderWithTheme-mount.html">632</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">521</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">527</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">630</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">605</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/InfoButton-mount_base.html">505</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/InfoButton-mount.html">574</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/MakeStyles-mount_base.html">1832</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/MakeStyles-mount.html">1714</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/Persona-mount_base.html">2925</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/Persona-mount.html">2706</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/SpinButton-mount_base.html">2462</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26444/perf-test-react-components/SpinButton-mount.html">2448</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-combobox/src/components/Combobox/useCombobox.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import * as React from 'react';
+import * as keys from '@fluentui/keyboard-keys';
```
[the remark this answers is not in the document]
Comment (smhigley) [reply packages/react-components/react-combobox/src/components/Combobox/useCombobox.tsx]: done!

**Pull request merged**